internal/fs/fs.go

```diff
@@ -2248,30 +2248,38 @@ func (fs *fileSystem) Unlink(
 		ctx, cancel = util.IsolateContextFromParentContext(ctx)
 		defer cancel()
 	}
-	// Find the parent.
+
 	fs.mu.Lock()
+
+	// Find the parent and file name.
 	parent := fs.dirInodeOrDie(op.Parent)
+	fileName := inode.NewFileName(parent.Name(), op.Name)
+
+	// Fetch the inode.
```

Comment:
can you add a comment here why we are only checking localFileInodes and generationBackedInodes.